From af11a3a2c0e764de2d551362d02109428ab79ca8 Mon Sep 17 00:00:00 2001
From: Simon Gaiser <simon@invisiblethingslab.com>
Date: Wed, 11 Apr 2018 22:02:08 +0200
Subject: [PATCH v2 1/2] x86/microcode: Indicate "not found" in rc of
 microcode_resume_cpu()

Make it possible to distinguish between a failure to load a microcode
update and a failure to find any matching microcode update by returning
-ENOENT (instead of -EIO) in the later case.

Signed-off-by: Simon Gaiser <simon@invisiblethingslab.com>
---
 xen/arch/x86/microcode.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xen/arch/x86/microcode.c b/xen/arch/x86/microcode.c
index 77c1efc97f..e9fafb1d14 100644
--- a/xen/arch/x86/microcode.c
+++ b/xen/arch/x86/microcode.c
@@ -248,7 +248,7 @@ int microcode_resume_cpu(unsigned int cpu)
     __microcode_fini_cpu(cpu);
     uci->cpu_sig = nsig;
 
-    err = -EIO;
+    err = -ENOENT;
     for_each_online_cpu ( cpu2 )
     {
         uci = &per_cpu(ucode_cpu_info, cpu2);
-- 
2.16.2

